### DISCUSSION TITLE
Suggestion - Update docs to add service users to /etc/group gpio group

### DISCUSSION THREAD

**Discussion by jhuettner** [CrazyIvan359/mqttany#141] [Troubleshooting] [answered]
Great project BTW and fantastic documentation!!!  Just a suggestion, I followed the installation guide for setup.  I was unsure if the systemd service needed a dedicated user or not.  I went ahead and created a user and group named mqttany.  I updated the /etc/systemd/system/mqttany.service file to set User=mqttany and Group=mqttany.  I then tried to start the service.  Service would not start.  journalctrl -u mqttany showed the following error:

2022-03-11 09:46:48,501 [INFO ] [core.gpio               ] Detected board:
Mar 11 09:46:48 hbpi03 python3[993]: 2022-03-11 09:46:48,514 [ERROR] [core                    ]
Mar 11 09:46:48 hbpi03 python3[993]: Traceback (most recent call last):
Mar 11 09:46:48 hbpi03 python3[993]:   File "mqttany.py", line 110, in <module>
Mar 11 09:46:48 hbpi03 python3[993]:     core.start(queue_core, args.config_file)
Mar 11 09:46:48 hbpi03 python3[993]:   File "/usr/local/mqttany/mqttany/core.py", line 384, in start
Mar 11 09:46:48 hbpi03 python3[993]:     gpio.init()
Mar 11 09:46:48 hbpi03 python3[993]:   File "/usr/local/mqttany/mqttany/gpio/__init__.py", line 82, in init
Mar 11 09:46:48 hbpi03 python3[993]:     os.getlogin(),
Mar 11 09:46:48 hbpi03 python3[993]: OSError: [Errno 6] No such device or address
Mar 11 09:46:48 hbpi03 python3[993]:
Mar 11 09:46:48 hbpi03 python3[993]: 2022-03-11 09:46:48,521 [WARN ] [core                    ] MQTTany exiting
Mar 11 09:46:48 hbpi03 systemd[1]: mqttany.service: Main process exited, code=exited, status=1/FAILURE
Mar 11 09:46:48 hbpi03 systemd[1]: mqttany.service: Failed with result 'exit-code'.
Mar 11 09:46:58 hbpi03 systemd[1]: mqttany.service: Service RestartSec=10s expired, scheduling restart.
Mar 11 09:46:58 hbpi03 systemd[1]: mqttany.service: Scheduled restart job, restart counter is at 1.

   After looking at the __init__.py file it appears the problem was that the user I setup did not have the ability to read the gpio pins.  I solved the problem by editing /etc/group to add my mqttany user to the gpio group.  After that I restarted the service and it runs fine now.  Not sure why os.getlogin() fails, but might suggest updating docs to add user to that group.  Running an Raspberry Pi 4 on Raspbian 10 (buster) fully up-to-date.  Also documenting to help anyone else with this problem.  Great project, thank you!

**Comment by CrazyIvan359** [comment, marked as answer]
Hi @jhuettner sorry it took so long to respond to this! The first half of this year wasn't great for me and this must have slipped through the cracks. Thank you for your kind words.

It is up to you if you want to create a dedicated user, but MQTTany is happy to run as any user.

The need for permissions is documented in each module's documentation as different permissions may be needed for different modules. This is indicated at the top of the [GPIO](https://crazyivan359.github.io/mqttany/interface/gpio/index.html#permissions) page. 

That said, however, the call to `os.getlogin()` is meant only to return the user MQTTany is running as for the log message that says you don't have permissions. I'm not sure why it failed but that is a bug and warrants investigation. I have created an issue from this discussion, can you pop over there and give me a bit more information (if you can still remember)?
